drivers/builtin/include/mbedtls/nist_kw.h

```diff
@@ -38,118 +38,66 @@ typedef enum {
 } mbedtls_nist_kw_mode_t;
 
 /**
- * \brief    The key wrapping context-type definition. The key wrapping context is passed
- *           to the APIs called.
- *
- * \note     The definition of this type may change in future library versions.
- *           Don't make any assumptions on this context!
- */
-typedef struct {
-    mbedtls_cipher_context_t MBEDTLS_PRIVATE(cipher_ctx);    /*!< The cipher context used. */
-} mbedtls_nist_kw_context;
-
-/**
- * \brief           This function initializes the specified key wrapping context
- *                  to make references valid and prepare the context
- *                  for mbedtls_nist_kw_setkey() or mbedtls_nist_kw_free().
+ * \brief                    This function encrypts a buffer using key wrapping.
  *
- * \param ctx       The key wrapping context to initialize.
+ * \param key                The key wrapping PSA key ID to use for encryption. The key should have the following attributes:
+ *                               - type: #PSA_KEY_TYPE_AES
+ *                               - algorithm: #PSA_ALG_ECB_NO_PADDING
+ *                               - usage flag: #PSA_KEY_USAGE_ENCRYPT + other flags if required by the application.
+ * \param mode               The key wrapping mode to use (#MBEDTLS_KW_MODE_KW or #MBEDTLS_KW_MODE_KWP)
+ * \param input              The buffer holding the input data.
+ * \param input_length       The length of the input data in Bytes.
+ *                           The input uses units of 8 Bytes called semiblocks.
+ *                           <ul><li>For KW mode: a multiple of 8 bytes between 16 and 2^57-8 inclusive. </li>
+ *                           <li>For KWP mode: any length between 1 and 2^32-1 inclusive.</li></ul>
+ * \param[out] output        The buffer holding the output data.
+ *                           <ul><li>For KW mode: Must be at least 8 bytes larger than \p in_len.</li>
+ *                           <li>For KWP mode: Must be at least 8 bytes larger rounded up to a multiple of
+ *                           8 bytes for KWP (15 bytes at most).</li></ul>
+ * \param[out] output_size   The number of bytes written to the output buffer. \c 0 on failure.
+ * \param[in] output_length  The capacity of the output buffer.
```

Comment:
* size and length are swapped.
* size isn't an `[in]`: that annotation is just for pointers.
* The documentation for length is missing the information that this is about the value stored at that address when the function returns.

```suggestion
 * \param output_size        The capacity of the output buffer.
 * \param[out] output_length On success, the number of bytes written to the output buffer. \c 0 on failure.
```